src/main/java/com/mewna/catnip/shard/event/DefaultDispatchManager.java

```diff
@@ -27,16 +27,112 @@
 
 package com.mewna.catnip.shard.event;
 
-import io.vertx.core.eventbus.MessageConsumer;
+import com.mewna.catnip.Catnip;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Scheduler.Worker;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.Accessors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.function.Consumer;
 
 public class DefaultDispatchManager extends AbstractDispatchManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultDispatchManager.class);
+    private final Map<String, Set<DefaultMessageConsumer<?>>> consumers = new ConcurrentHashMap<>();
+    private Worker worker;
+    
+    @Override
+    public void catnip(final Catnip catnip) {
+        super.catnip(catnip);
+        worker = catnip().rxScheduler().createWorker();
```

Comment:
To make sure I understand -- this is for the consistency thing we talked about before, right?

Reply:
No, the consistency thing we talked about before got replaced by Workers, which run in serial.

Reply:
:ok_hand: